contracts/btc-finality/src/events.rs

```diff
@@ -0,0 +1,42 @@
+use cosmwasm_std::Event;
+
+/// FinalityProviderStatus represents the status of a finality provider,
+/// following the same enum values as defined in babylon/proto/babylon/btcstaking/v1/events.proto
+/// ref: https://github.com/babylonlabs-io/babylon/blob/e01948b4919f04bae51e8910a93769132382ed13/proto/babylon/btcstaking/v1/events.proto#L129
+#[derive(Debug, Clone, PartialEq)]
+#[allow(dead_code)] // Some variants are not used yet but defined for completeness
+pub enum FinalityProviderStatus {
+    /// FINALITY_PROVIDER_STATUS_INACTIVE defines a finality provider that does not have sufficient
+    /// delegations or does not have timestamped public randomness.
+    Inactive,
+    /// FINALITY_PROVIDER_STATUS_ACTIVE defines a finality provider that have sufficient delegations
+    /// and have timestamped public randomness.
+    Active,
+    /// FINALITY_PROVIDER_STATUS_JAILED defines a finality provider that is jailed due to downtime
+    Jailed,
+    /// FINALITY_PROVIDER_STATUS_SLASHED defines a finality provider that is slashed due to double-sign
+    Slashed,
```

Comment:
We already have slashed events emitted https://github.com/babylonlabs-io/cosmos-bsn-contracts/blob/1775885936623bd400b1b1080f18a4a1680325bd/contracts/btc-finality/src/finality.rs#L381. Keeping the `Slashed` status is for completeness